### PR TITLE
#305/UI/form add edit delete contests

### PR DESCRIPTION
**Pull request by RobertS-ki**
Updated functionality.
All validations work now, updated icons, added button reaction on errors in validation. Button does not allow to submit, when no checkbox(classes) is selected. Converted from draft to genuine pull request.
obraz
obraz
obraz

### CODE REVIEW

File: client/src/Molecules/AddContestForm/AddContestForm.js [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+          </div>
+          {chosenClasses.length !== 0 && (
+            <>
+              <MainButton primary text="zarejestruj nowe zawody" />
+            </>
+          )}
+        </FormWrapper>
+        {chosenClasses.length === 0 && (
+          <>
+            <MainButton secondary disabled text="Dokończe rejestrację" />
```
Comment (n0macx) [inline client/src/Molecules/AddContestForm/AddContestForm.js:359]: Type - o

File: client/src/Atoms/SelectField/SelectFieldStyled.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  ${FONTS.body_semibold};
+  &::placeholder {
+    color: ${({ theme }) => theme.grey400};
+  }
+  &:focus {
+    border: 0.5px solid transparent;
+    outline: 3px solid ${({ theme }) => theme.primary201};
+  }
+`;
+
```
Comment (n0macx) [inline client/src/Atoms/SelectField/SelectFieldStyled.js:23]: Px to rem ?

File: client/src/Organisms/RegistrationForm/RegistrationFormSignup.js [hunk 3]
```diff
@@ -15,7 +14,8 @@ const RegistrationFormSignup = ({
   initialState,
   editPhoneNumber,
 }) => {
-  const { handleInputChange, submitHandler, formData, errors } = useForm(
+  // useForm name is reserved for react-hook-form
```
Comment (n0macx) [inline client/src/Organisms/RegistrationForm/RegistrationFormSignup.js:17]: I think its not necessary to leave the comment

**Review by n0macx — CHANGES_REQUESTED** [review]
Please take a look at comments.
Every field if errors could have a red outline.
Please change sort-imports default-sort-style to: renke.
Date format is Month - Day - Year - maybe it would be better to change it to day - month - year ? 
Maybe as a initial value dd/mm/yy ?

File: client/src/Molecules/AddContestForm/AddContestForm.js (hunk 1 above)
Comment (RobertS-ki) [reply client/src/Molecules/AddContestForm/AddContestForm.js:359]: Done.

File: client/src/Atoms/CheckBoxFormWrapper/CheckBoxFormWrapperStyled.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    label {
+      ${FONTS.caption};
+      height: 24px;
+      position: relative;
+      top: 3px;
+      padding: 0 0 0 10px;
+    }
+  }
+`;
+
```
[the remark this answers is not in the document]
Comment (RobertS-ki) [reply client/src/Atoms/CheckBoxFormWrapper/CheckBoxFormWrapperStyled.js:39]: Done.

File: client/src/Atoms/SelectField/SelectFieldStyled.js (hunk 2 above)
Comment (RobertS-ki) [reply client/src/Atoms/SelectField/SelectFieldStyled.js:23]: Done.

File: client/src/Molecules/AddContestForm/AddContestForm.js [hunk 5]
```diff
@@ -0,0 +1,367 @@
...
+              {chosenClasses.length === 0 && (
+                <p
+                  style={{
+                    position: 'absolute',
+                  }}
+                >
+                  Wybierz przynajmniej jedną klasę
+                </p>
+              )}
+            </CheckBoxFormWrapperStyled>
```
[the remark this answers is not in the document]
Comment (RobertS-ki) [reply client/src/Molecules/AddContestForm/AddContestForm.js:349]: Removed excessive props. Rest is ok :)

File: client/src/Molecules/CheckBoxFormWrapper/CheckBoxFormOuterWrapperStyled.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import styled from 'styled-components';
+
+import COLORS from './../../Styles/varsStyledComponents';
+
+const CheckBoxFormOuterWrapperStyled = styled.div`
+  /* border: 1px solid #7b8794; */
```
Comment (n0macx) [inline client/src/Molecules/CheckBoxFormWrapper/CheckBoxFormOuterWrapperStyled.js:6]: Could it be erased ?

Comment (RobertS-ki) [reply client/src/Molecules/CheckBoxFormWrapper/CheckBoxFormOuterWrapperStyled.js]: Done.

File: client/src/Organisms/RegistrationForm/RegistrationFormSignup.js (hunk 3 above)
Comment (RobertS-ki) [reply client/src/Organisms/RegistrationForm/RegistrationFormSignup.js]: Removed.

**Pull request merged**